File: app/src/main/java/com/obrockmole/betterdining/utils/Logger.java
package com.obrockmole.betterdining.utils;

import android.util.Log;

public class Logger {
    private static LogLevel logLevel = LogLevel.MINIMAL;

    public static void setLogLevel(String level) {
        try {
            logLevel = LogLevel.valueOf(level.toUpperCase());
        } catch (IllegalArgumentException e) {
            Log.e("Logger", "Invalid log level: " + level + ". " + e.getMessage());
            logLevel = LogLevel.MINIMAL;
        }
    }

    public static void LogInfo(String tag, String message) {
        if (logLevel == LogLevel.MINIMAL || logLevel == LogLevel.FULL) {
            Log.i(tag, message);
        }
    }

    public static void LogDebug(String tag, String message) {
        if (logLevel == LogLevel.FULL) {
            Log.d(tag, message);
        }
    }

    public static void LogWarning(String tag, String message) {
        if (logLevel == LogLevel.MINIMAL || logLevel == LogLevel.FULL) {
            Log.w(tag, message);
        }
    }

    public static void LogWarning(String tag, String message, Throwable throwable) {
        if (logLevel == LogLevel.MINIMAL || logLevel == LogLevel.FULL) {
            Log.w(tag, message, throwable);
        }
    }

    public static void LogError(String tag, String message) {
        if (logLevel == LogLevel.MINIMAL || logLevel == LogLevel.FULL) {
            Log.e(tag, message);
        }
    }

    public static void LogError(String tag, String message, Throwable throwable) {
        if (logLevel == LogLevel.MINIMAL || logLevel == LogLevel.FULL) {
            Log.e(tag, message, throwable);
        }
    }
}

File: sharedLogic/src/commonMain/kotlin/com/obrockmole/betterdining/utils/Logger.kt
package com.obrockmole.betterdining.utils

expect fun writePlatformLog(logLevel: LogLevel, tag: String, message: String)

object Logger {
    var logAmount = LogAmount.MINIMAL

    fun setLogAmount(amount: String) {
        try {
            logAmount = LogAmount.valueOf(amount.uppercase())
        } catch (e: IllegalArgumentException) {
            writePlatformLog(LogLevel.ERROR, "Logger", "Invalid log level: " + amount + ". " + e.message)
            logAmount = LogAmount.MINIMAL
        }
    }

    fun LogInfo(tag: String, message: String) {
        if (logAmount == LogAmount.MINIMAL || logAmount == LogAmount.FULL) {
            writePlatformLog(LogLevel.INFO, tag, message)
        }
    }

    fun LogDebug(tag: String, message: String) {
        if (logAmount == LogAmount.FULL) {
            writePlatformLog(LogLevel.DEBUG, tag, message)
        }
    }

    fun LogWarning(tag: String, message: String) {
        if (logAmount == LogAmount.MINIMAL || logAmount == LogAmount.FULL) {
            writePlatformLog(LogLevel.WARN, tag, message)
        }
    }

    fun LogError(tag: String, message: String) {
        if (logAmount == LogAmount.MINIMAL || logAmount == LogAmount.FULL) {
            writePlatformLog(LogLevel.ERROR, tag, message)
        }
    }
}